### PR TITLE
Expose Tail Kind Call to MLIR

### PR DESCRIPTION
**Pull request by stumpOS**
I would like to mark a call op in LLVM dialect as Musttail. The calling convention attribute only exposes Tail, not Musttail. I noticed that the CallInst of LLVM has an additional field to specify the flavor of tail call kind. I bubbled this up to the LLVM dialect by adding another attribute that maps to LLVM::CallInst::TailCallKind.

### CODE REVIEW

File: mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td [hunk 1]
```diff
@@ -1077,4 +1077,14 @@ def LLVM_PoisonAttr : LLVM_Attr<"Poison", "poison">;
 /// Folded into from LLVM::ZeroOp.
 def LLVM_ZeroAttr : LLVM_Attr<"Zero", "zero">;
 
+
```
Comment (Mogball) [inline mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td:1080]: style: extra newline

**Review by Mogball — APPROVED** [review]

File: mlir/include/mlir/IR/DialectInterface.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include "mlir/Support/TypeID.h"
 #include "llvm/ADT/DenseSet.h"
 #include "llvm/ADT/STLExtras.h"
+#include <vector>
```
Comment (Mogball) [inline mlir/include/mlir/IR/DialectInterface.h:15]: please remove this

**Review by gysit — COMMENTED** [review]
Thanks for adding the attribute.

If you can add the attribute to the import as well that would be great!

It would require an additional line here:
https://github.com/llvm/llvm-project/blob/2217933c6d3850948dced70097847f5a684ed069/mlir/lib/Target/LLVMIR/ModuleImport.cpp#L1470
and a test file in the same folder as:
https://github.com/llvm/llvm-project/blob/bb8998dd3b74a6936452d7ddeb341e79c1d8f001/mlir/test/Target/LLVMIR/Import/calling-convention.ll#L1
You can use the exported result of your export test to produce an import test.

**Comment by stumpOS** [comment]
> Thanks for adding the attribute.
> 
> If you can add the attribute to the import as well that would be great!
> 
> It would require an additional line here:
> 
> https://github.com/llvm/llvm-project/blob/2217933c6d3850948dced70097847f5a684ed069/mlir/lib/Target/LLVMIR/ModuleImport.cpp#L1470
> 
> and a test file in the same folder as:
> 
> https://github.com/llvm/llvm-project/blob/bb8998dd3b74a6936452d7ddeb341e79c1d8f001/mlir/test/Target/LLVMIR/Import/calling-convention.ll#L1
> 
> You can use the exported result of your export test to produce an import test.

I responded to the suggestion @gysit ! Let me know if it looks good to you

File: mlir/test/Target/LLVMIR/Import/tail-kind.ll [hunk 3]
```diff
@@ -0,0 +1,38 @@
+; RUN: mlir-translate -import-llvm -split-input-file %s | FileCheck %s
+
+; CHECK: llvm.func @tailkind()
+declare void @tailkind()
+
+; CHECK-LABEL: @call_tailkind
+define void @call_tailkind() {
+  ; CHECK: llvm.call musttail @tailkind()
+  musttail call void @tailkind()
+
```
Comment (gysit) [inline mlir/test/Target/LLVMIR/Import/tail-kind.ll:10]:
```suggestion
```
ultra nit: I would drop these newlines before the return here and in the tests below.

**Review by gysit — APPROVED** [review]
Thank for adding the import!

LGTM modulo nit.

**Pull request merged**